R/ui_functions.R

```diff
@@ -612,3 +612,110 @@ ui_cp_download_single <- function(country,
 
   return(out)
 }
+
+#' Country Profile Key Indicators download
+#'
+#' Helper function to download Country Profile data
+#'
+#' @inheritParams ui_cp_download
+#' @return list
+#' @keywords internal
+ui_cp_download_multiple <- function(country,
+                                    povline = 1.9,
+                                    lkup,
+                                    indicators = "all") {
+
+
+  ### get indicator list of interest
+  ind_table <- pip(country = country,
+                   povline = povline,
+                   lkup = lkup,
+                   welfare_type = "all",
+                   reporting_level = "all")
```

Comment:
The `pip` function is already being called by `ui_cp_charts()` to populate the country profile page. I think we should recycle those calls, otherwise we are just making the same calls multiple time, and this will be very inefficient.

Reply:
Are you suggesting that I run the ui_cp_charts() function again instead of calling pip()? It isn't clear how to write the function without having to re-run pip either directly or through ui_cp_charts() in the function I am creating. Also, ui_cp_charts() is creating other elements which ui_cp_download_multiple() (the function I am writing) doesn't seem to need so running ui_cp_charts() takes significantly longer to run than simply calling pip again.

Reply:
Hi @ifeanyi588 We currently have two functions that feed the country profile page:
- `ui_cp_charts` returns a list containing all the elements to populate the charts
- `ui_cp_key_indicators` returns all the data points to populate the "Key indicators (most recent value)" section

Both functions call the `pip()` function at one point or the other. I think you new function should at least be combined with the `ui_cp_key_indicators` function so at the highest level, only the `ui_cp_key_indicators` function is being called, and all the necessary data points are being returned by this high-level function.

Further optimization would imply merging both `ui_cp_charts` and `ui_cp_key_indicators` functions. But maybe we'll keep that for a next iteration.

Reply:
@tonyfujs I am looking into a modification of the `ui_cp_key_indicators` function that incorporates everything `ui_cp_download_multiple` is planned to return. I noticed that `ui_cp_key_indicators` already returns pretty much the same thing except that headcount rates are fixed the only produce latest year numbers while we would need the history. Would it be ok to modify ui_cp_key_indicators() to do this?